src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -506,12 +496,58 @@ internal void TryDisconnectExistingDeferredProject(IVsHierarchy hierarchy, strin
             }
         }
 
-        private bool IsDeferredSolutionLoadEnabled()
+        public void OnBeforeCloseSolution()
         {
-            // NOTE: It is expected that the "as" will fail on Dev14, as IVsSolution7 was
-            // introduced in Dev15.  Be sure to handle the null result here.
-            var solution7 = _serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution7;
-            return solution7?.IsSolutionLoadDeferred() == true;
+            AssertIsForeground();
+
+            _solutionIsClosing = true;
+
+            foreach (var p in this.ImmutableProjects)
+            {
+                p.StopPushingToWorkspaceHosts();
+            }
+
+            _solutionLoadComplete = false;
+
+            // Cancel any background solution parsing. NOTE: This means that work needs to
+            // check the token periodically, and whenever resuming from an "await"
+            _solutionParsingCancellationTokenSource.Cancel();
+            _solutionParsingCancellationTokenSource = new CancellationTokenSource();
+        }
+
+        public void OnAfterCloseSolution()
+        {
+            AssertIsForeground();
+
+            if (VisualStudioWorkspaceImpl.IsDeferredSolutionLoadEnabled(_serviceProvider))
```

Comment:
Do you want to store the value returned from this method in OnBeforeCloseSolution() and use it here? That way you have the correct value for the solution which just closed.
If the global and the solution-level setting of LSL are not the same, then the value returned here will be incorrect.

Please verify that scenario.